### ISSUE TITLE
ios-XCBuildConfiguration-CODE_SIGN_IDENTITY not respecting buildType attribute in latest versions of cordova-ios platform

### ISSUE THREAD

**Issue by erichochhalter** [dpa99c/cordova-custom-config#28]
In platform cordova-ios v4.0+, `build.xcconfig` already contains Developer entries for `CODE_SIGN_IDENTITY` which get incorrectly clobbered by custom configuration tags with `buildType="release"`.

While I haven't encountered broken builds from this behavior, I get very nervous when the value `iPhone Developer` in `build.xcconfig` gets clobbered by my distribution code sign identity value. I expect my distribution code sign identify string to be added only to `build-release.xcconfig`. I expect my developer code sign identity string (set by a `buildType="debug"` preference) to be what is added to `build.xcconfig` (I think? Or would it make more sense for the value in `build.xcconfig` be removed and moved to `build-debug.xcconfig`?). I wonder if this behavior where `buildType="release"` preferences are getting applied to both `build-release.xcconfig` and `build.xcconfig` for any other XCBuildConfiguration keys?

**cordova-ios v4.0 build.xcconfig and build-release.xcconfig**
https://github.com/apache/cordova-ios/blob/4.0.0/bin/templates/scripts/cordova/build.xcconfig
https://github.com/apache/cordova-ios/blob/master/bin/templates/scripts/cordova/build-release.xcconfig

---

**build.xcconfig** before `cordova prepare ios --release --device`:
```
CODE_SIGN_IDENTITY = iPhone Developer
CODE_SIGN_IDENTITY[sdk=iphoneos*] = iPhone Developer
```

**config.xml**:
```
<preference name="ios-XCBuildConfiguration-CODE_SIGN_IDENTITY" value="iPhone Developer" buildType="debug" quote="none"/>
<preference name="ios-XCBuildConfiguration-CODE_SIGN_IDENTITY" value="iPhone Distribution: {{MY_DISTIBUTION_CODE_SIGN_IDENTITY}}" buildType="release" quote="none"/>
```

**build.xcconfig** after `cordova prepare ios --release --device`:
```
CODE_SIGN_IDENTITY = iPhone Distribution: {{MY_DISTIBUTION_CODE_SIGN_IDENTITY}}
CODE_SIGN_IDENTITY[sdk=iphoneos*] = iPhone Developer
```

---

**build-release.xcconfig** before `cordova prepare ios --release --device`:
```
CODE_SIGN_IDENTITY = iPhone Distribution
CODE_SIGN_IDENTITY[sdk=iphoneos*] = iPhone Distribution
```

**config.xml**:
```
<preference name="ios-XCBuildConfiguration-CODE_SIGN_IDENTITY" value="iPhone Developer" buildType="debug" quote="none"/>
<preference name="ios-XCBuildConfiguration-CODE_SIGN_IDENTITY" value="iPhone Distribution: {{MY_DISTIBUTION_CODE_SIGN_IDENTITY}}" buildType="release" quote="none"/>
```

**build-release.xcconfig** after `cordova prepare ios --release --device`:
```
CODE_SIGN_IDENTITY = iPhone Distribution: {{MY_DISTIBUTION_CODE_SIGN_IDENTITY}}
CODE_SIGN_IDENTITY[sdk=iphoneos*] = iPhone Developer
```

**Comment by dpa99c** [comment]
This should be fixed in commit 584640651a4b615f1b9b979e51a3282cfb80e537.

I've added some special behaviour for CODE_SIGN_IDENTITY, but this can be overridden using the new `xcconfigEnforce` attribute. This has been published to npm as v1.2.0. Let me know if you have any issues.

**Comment by dpa99c** [comment]
Should be fixed - reopen if not
